mesa: update to 25.0.1

zlice opened the pull request:
#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, **x86_64**

#### Comments

libglapi moved into gallium https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/32789

glxinfo and glxgears work fine. tested kotor which is opengl, tried portal with `-gl` but i'm not sure if it was using opengl or not, kept trying to build vulkan shaders (A770)

zlice commented:
looks like a repo or something not sync'd ? `libwayland-egl` was moved to `wayland` https://github.com/void-linux/void-packages/commit/69b59e3eac260e8d79528b36daf4c7d61c5b5ed6

similar to `mlt` for the ffmpeg-meta pr

oreo639 commented:
> similar to `mlt` for the ffmpeg-meta pr

When packages are added to removed-packages, they don't immediately get removed from the repos, that has to be done manually.

```
$ XBPS_TARGET_ARCH=x86_64-musl xbps-query -MRs libwayland-egl -p run_depends --repository=https://repo-default.voidlinux.org/current/musl
wlc-0.0.11_1: libwayland-egl>=9.0.1_4 (https://repo-default.voidlinux.org/current/musl)
```

zlice commented:
thx, figured it was something like that

tm512 commented:
For what it's worth, I've been testing this PR for the last week, and it's been smooth sailing here on an RX 6650 XT.

shara-k commented:
Both `libxml2` and `ncurses` were updated within the last week; when I attempted to build this PR today, xbps-src failed due to conflicting packages being present within the masterdir

```
=> ERROR: libxml2-2.13.6_1: failed to install target dependencies! (error 19)
ncurses-libs-6.5_1 in transaction breaks installed pkg 'ncurses-6.5_2'
Transaction aborted due to unresolved dependencies.
```

zlice commented:
just had a hard system freeze watching a video that got choppy when it shouldn't really be. but also updated intel-media-drivers. rolled this back to test

zlice commented:
went back to 25.0.3

had another crash with 25.0.4 while playing a game. was at least able to ssh and restart but everything has been fine on 25.0.3

ericonr commented:
Have you managed to test https://docs.mesa3d.org/relnotes/25.0.5.html ?

zlice commented:
> Have you managed to test https://docs.mesa3d.org/relnotes/25.0.5.html ?

no. no one replied to my issue about the crashing. only see 1 thing in there that may be related to it so it wasnt a high priority. rather have this merged since it's working for ppl.

chilledfrogs commented:
Alpine merged 25.0.5 recently, so I'd say that it would be worth a check actually

zlice commented:
well so far so good for 25.0.5. '4 was crapping out in hours and it's been about all day

ericonr commented on srcpkgs/mesa/template:
```diff
@@ -23,8 +23,9 @@ maintainer="Orphaned <orphan@voidlinux.org>"
 changelog="https://docs.mesa3d.org/relnotes.html"
-distfiles="https://mesa.freedesktop.org/archive/mesa-${version}.tar.xz"
-checksum=e641ae27191d387599219694560d221b7feaa91c900bcec46bf444218ed66025
+#distfiles="https://mesa.freedesktop.org/archive/mesa-${version}.tar.xz"
+distfiles="https://mesa3d.org/archive/mesa-${version}.tar.xz"
```
Why did you change the distfiles URL?

ericonr replied:
Ok, tested it here. The old link can simply be removed entirely, no reason to leave it commented out.

ericonr reviewed (commented):
> libglapi moved into gallium https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/32789

I think it would be nice to include this info in the commit message :)

zlice commented:
> > libglapi moved into gallium https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/32789
> 
> I think it would be nice to include this info in the commit message :)

added this. removed the commented distfile since it permanently moved

zlice commented:
well, i was fine for a day and then updated discord and started getting crashes and oom today. turned off hwaccel launching discord to see if that helps anything. closed my mesa issue which may have been premature. if anyone else has issues with 25.0.5 or similar freezes/oom where you need to hold power to reboot lmk.

zlice commented:
moved back to 25.0.3 and haven't had any issues with all of the same things open (browser and discord are really it sitting there while i made food).

reopened my mesa issue. there seems to be something wrong with 25.0.4 at least for Intel GPUs

LangleyAsuka commented:
Bringing any version of mesa 25.X into the repos would be great, as especially mesa 25.0.0 had some crucial fixes for AMD cards. There were some issues with especially Chromium programs but also random other graphical applications leading to random freezes. There was also a whole discussion on the Arch forums of mesa 24.3.X being the bunch of versions with the most issues regarding this matter.

zlice commented:
ah so they fixed 1 set of cards and fuked another lol. great

zlice commented:
@dkwo lmk about the asahi stuff, if you wana do that with or after this

classabbyamp commented:
the musl patches might be able to be made conditional to `"$XBPS_TARGET_WORDSIZE" -eq 32 -a "$XBPS_TARGET_LIBC" = musl`

classabbyamp commented on srcpkgs/removed-packages/template:
```diff
@@ -353,6 +353,7 @@ replaces="
  libArcus<=4.13.1_8
  libCharon<=4.13.0_3
  libOSMesa-asahi<=25.1.0+20250221_1
+ libOSMesa<=25.1.0
```
also should be `<=24.3.4_2`

zlice commented on srcpkgs/mesa/template:
```diff
@@ -81,6 +81,7 @@ esac
 if [ "$_have_llvmpipe" ]; then
 	subpackages+=" mesa-vulkan-lavapipe"
 	_vulkan_drivers+=",swrast"
+  # note: swrast IS valid for vulkan drivers still, not gallium
```
usually that's picked up by xlint, know why it wasn't?

zlice commented:
> the musl patches might be able to be made conditional to `"$XBPS_TARGET_WORDSIZE" -eq 32 -a "$XBPS_TARGET_LIBC" = musl`

think i got everything you mentioned except this, not sure what you meant

classabbyamp commented:
armv*-musl are still on musl 1.1, so they will fail to build

chibiace commented:
ive been on 25.1.1 for a week with my 9070xt, havent had any problems. learnt alot about how to build packages for void too.

LangleyAsuka commented:
I have now been using mesa 25.1.1 for a couple of days now and it seems to work perfectly fine on my Rx 6800.
The freezes I was talking about earlier were caused by a regression in the kernel which has now been patched so it has been smooth sailing here for me.
Are there any further roadblocks for getting this merged? So for example users with a Rx 9XXX series card will have a decent experience?

zlice commented:
ya i saw 2 botched and 3 fixed but phoronix bit makes it sound it it was a fix just a bad 1

`25.1.1 = cf942a18b7b9e9b88524dcbf0b31fed3cde18e6d52b3375b0ab6587a14415bce`

zlice commented:
ik you changed metapackage string recently. whys asahi need **_**metapackage ?

classabbyamp commented on srcpkgs/mesa-asahi/template:
```diff
@@ -1,168 +1,64 @@
-replaces="mesa>=0 mesa-libgallium>=0 libglapi>=0"
-provides="mesa-${_mesaver}_${revision} mesa-libgallium-${_mesaver}_${revision} libglapi-${_mesaver}_${revision}"
-shlib_provides="libglapi.so libgallium-${_mesaver}-devel.so"
+_metapackage=yes
```
xlint is wrong atm, `metapackage=yes` is correct

https://github.com/leahneukirchen/xtools/pull/340

classabbyamp commented on srcpkgs/removed-packages/template:
```diff
@@ -2,11 +2,11 @@
 maintainer="Piotr Wójcik <chocimier@tlen.pl>"
 license="Public Domain"
 homepage="https://voidlinux.org"
+metapackage=yes
```
it was in the correct place already
